crates/middleware/src/thesaurus/mod.rs

```diff
@@ -288,6 +286,19 @@ fn index_inner(name: String, messages: Vec<Message>) -> Thesaurus {
                 for synonym in synonyms {
                     let nterm = NormalizedTerm::new(concept.id.clone(), synonym.into());
                     thesaurus.insert(concept.value.clone(), nterm);
+
+                    // Create a document (`Document`) for each synonym
+                    todo!("Do we want to create a document for each synonym?")
```

Comment:
@AlexMikhalev what do you think? I think we'll need to create the documents when we build the `KnowledgeGraph` with `logseq` here, right?

Otherwise, we won't have an index to retrieve the documents from later.

Unless I'm misunderstaning something.

Reply:
When we create a thesaurus, we are only interested in terms; creating a document from concept is optional when reading data for a knowledge graph: it will consume more memory, and there is currently no interface to display it, but adding a display modal when clicking on the tag is easy enough on the front end. At best, we only need a single document for a single term - synonyms are referencing concepts via ID.